### PR TITLE
Remove TitlePart wrapper when it is hidden

### LINKED ISSUE #15382: Remove TitlePart wrapper when it is hidden
When adding a title part on a content part and you configure the `TitlePart` as hidden, the UI render the following HTML

```
<div class="row mb-3 content-part-wrapper content-part-wrapper-title-part">


</div>
```

It should not create the wrapper if the title is hidden.

### PR DESCRIPTION
**Pull request by douwinga**
Fixes #15382

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Title/Views/ContentPart-TitlePart.Edit.cshtml
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Title/Views/ContentPart-TitlePart.Edit.cshtml:20]: Add new line

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

**Comment by sebastienros** [comment]
@douwinga PR is fine, did you actually face an issue with this empty wrapper? Or did you find the ticket and wanted to help fix it.

**Review by sebastienros — APPROVED** [review]

**Comment by douwinga** [comment]
@sebastienros Just trying to contribute to become a better developer. Was looking for a ticket I could finish on my lunch break. This was the first one I found.

**Comment by sebastienros** [comment]
So we are hoping this doesn't break anything, the issue wasn't to fix an actual problem but something that looked odd. It's totally possible that some other feature was using (or could use in the future) these wrappers even if the shape was not rendered (e.g. content preview, placement, shape debugger, ...)

**Comment by douwinga** [comment]
Hmmm, I bet this change would actually cause an issue with admin templates and shape rendering. It would no longer obey the hide check. Should probably keep that check on the TitlePart.Edit. it seemed like an innocent change

**Comment by sebastienros** [comment]
If you can do some test and see if nothing is broken, otherwise we'll just revert the change.

**Comment by sebastienros** [comment]
> I wonder if these wrappers were just to give JavaScript something to target or something.

Definitely. Question is do these wrappers still like to have the div even if the part is not displayed ... I would say that if the wrapper doesn't bother, we should just keep it.

**Comment by MikeAlhayek** [comment]
I tested the change before merging and it worked as expected with no issues. But another set of eyes wont hurt.

**Comment by douwinga** [comment]
I was wrong. Having `{{ Model.Parts.TitlePart | shape_render }}` in an admin template does still obey the hidden setting. So far nothing else has broken either.

**Comment by douwinga** [comment]
It breaks with a placement.json with the following:

```
{
  "TitlePart_Edit": [
    {
      "place": "Parts:0",
      "displayType": "Edit"
    }
  ]
}
```
If we do the `TitlePartOptions.GeneratedHidden` check in `TitlePart.Edit.cshtml` it works correctly as the only thing that is broken is that it no longer is checking that.

I can do a PR to just have it check in `TitlePart.Edit.cshtml` too and not just `ContentPart-TitlePart.Edit` if we think we are not just going to revert this change.

**Comment by douwinga** [comment]
> shape debugger

Not sure what that is to test

**Pull request merged**
